src/fanout/emqx_sdv_fanout_data.erl

```diff
@@ -0,0 +1,58 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2025 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+-module(emqx_sdv_fanout_data).
+
+-export([
+    create_tables/0,
+    exists/1,
+    insert/2,
+    read/1
+]).
+
+-include("emqx_sdv.hrl").
+
+%% @doc Create the tables.
+create_tables() ->
+    ok = mria:create_table(?DATA_TAB, [
+        {type, ordered_set},
+        {rlog_shard, ?DB_SHARD},
+        {storage, ram_copies},
+        {record_name, ?DATA_REC},
+        {attributes, record_info(fields, ?DATA_REC)}
+    ]).
+
+%% @doc Hot path, do not insert unless it's new.
+insert(Ts, Data) ->
+    Sha1 = crypto:hash(sha, Data),
+    case exists(Sha1) of
+        true ->
+            ok;
+        false ->
+            mria:dirty_write(?DATA_TAB, #?DATA_REC{id = Sha1, ts = Ts, data = Data})
+    end,
+    {ok, Sha1}.
+
+%% @doc Hot path, avoid reading body from disk when checking its existence.
+exists(Sha1) ->
+    case mnesia:dirty_next(?DATA_TAB, seudo_prev(Sha1)) of
+        Sha1 ->
+            true;
+        _ ->
+            false
+    end.
+
+%% @doc Read the data from disk.
```

Comment:
I don't quite get this comment: ?DATA_TAB is a `ram_copies` table so nothing is read from disk.

Reply:
thanks. will change to disc_copies

Reply:
It will still read from RAM, so I suggest to also change the comment to "read the data from table"

Reply:
it was previously rocksdb_copies, changed mind to use disc_copies, but for some reason, typed ram_copies